client/authority-discovery/src/worker.rs

```diff
@@ -374,9 +374,11 @@ where
 			.map_err(|e| Error::CallingRuntime(e.into()))?
 			.into_iter()
 			.filter(|id| !local_keys.contains(id.as_ref()))
-			.collect();
+			.collect::<Vec<_>>();
 
-		self.addr_cache.retain_ids(&authorities);
+		// Inform the address cache about the new maximum number of
+		// addresses to cache.
+		self.addr_cache.set_max_authority_ids(authorities.len());
```

Comment:
I don't quite get why we use lru instead of `retain_ids`? What are the benefits? I don't see it's harder to implement multiple authority ids per peer_id with the previous API.

Reply:
You were right that this was bullshit, I reverted it :D